src/createServer.js

```diff
@@ -1,11 +1,219 @@
 'use strict';
 
-// const express = require('express');
+const express = require('express');
 
 function createServer() {
-  // Use express to create a server
-  // Add a routes to the server
-  // Return the server (express app)
+  const app = express();
+
+  const users = {};
+  const expenses = {};
+
+  let nextUserId = 1;
+  let nextExpenseId = 1;
+
+  app.use(express.json());
+
+  app.post('/users', (req, res) => {
+    const { name } = req.body;
+
+    if (!name) {
+      return res.status(400).send('Name is required');
+    }
+
+    const user = {
+      id: nextUserId++,
+      name,
+    };
+
+    users[user.id] = user;
+
+    res.status(201).json(user);
+  });
+
+  app.get('/users/', (req, res) => {
+    res.status(200).json(Object.values(users));
+  });
+
+  app.get('/users/:id', (req, res) => {
+    const { id } = req.params;
+    const user = users[id];
+
+    if (!user) {
+      return res.status(404).send('User not found');
+    }
+
+    res.status(200).json(user);
+  });
+
+  app.patch('/users/:id', (req, res) => {
+    const { id } = req.params;
+    const user = users[id];
+
+    if (!user) {
+      return res.status(404).send('User not found');
+    }
+
+    const { name } = req.body;
+
+    if (!name) {
+      return res.status(400).send('Name is required');
+    }
+
+    user.name = name;
+
+    res.status(200).json(user);
+  });
+
+  app.delete('/users/:id', (req, res) => {
+    const { id } = req.params;
+    const user = users[id];
+
+    if (!user) {
+      return res.status(404).send('User not found');
+    }
+
+    delete users[id];
+
+    res.status(204).send();
+  });
+
+  app.post('/expenses', (req, res) => {
+    const { userId, spentAt, title, amount, category, note } = req.body;
+
+    if (!userId) {
+      return res.status(400).send('User ID is required');
+    }
+
+    if (!title) {
+      return res.status(400).send('Title is required');
```

Comment:
Do not rely on the unknown string, make constants for this.
const Errors = {
  ....
};